drafts/maui-test_pt1.md

```diff
@@ -0,0 +1,332 @@
+            {
+                Cats.Add(cat);
+            }
+        }
+        catch (Exception ex)
+        {
+            Debug.WriteLine(ex);
+            await Shell.Current.DisplayAlert("Error", "Unable to get cats", "Ok");
+        }
+        finally
+        {
+            IsBusy = false;
+        }
+    }
+```
+
+Sidenote: the method GetTenCatsAsync uses the method FileSystemCsvService under the hood.
+
+The following test verifies whether the method adds exactly ten cats to the list and whether the method CheckNetworkConnectivity calls the connectivity mock exactly once.
+
+``` C#
+
+    [Fact]
+    public async void GetCatsAsync()
+    {
+        var connectivtyMock = GetConnectivtyMock();
+        var geolocationMock = GetGeolocationMock();
+        var catServiceMock = GetCatServiceMock();
+        var catsViewModel = new CatsViewModel(catServiceMock.Object, connectivtyMock.Object, geolocationMock.Object);
+
+        await catsViewModel.GetCatsAsync();
+
+        const int expectedCatsCount = 10;
+        catsViewModel.Cats.Should().HaveCount(expectedCatsCount);
+        catServiceMock.Verify(catService => catService.GetTenCatsAsync(), Times.Once());
+        connectivtyMock.Verify(connectivity => connectivity.NetworkAccess, Times.Once());
+
+    }
+```
+
+The following test verifies whether the method adds exactly ten cats to the list and whether the method invokes CatServiceMock and ConnectivityMock once.
+
+Here is the second method of the view model:
+
+``` C#
+    [RelayCommand]
+    public async Task GoToCatDetailsAsync(Cat cat)
+    {
+        if (cat is null) return;
+
+        await Shell.Current.GoToAsync($"{nameof(DetailsPage)}", true, new Dictionary<string, object>
+        {
+            {$"{nameof(cat)}", cat}
+        });
+    }
+```
+
+The method uses the Shell to navigate to the details page. It passes down a cat object via a dictionary, which the details page can access.
+
+Let us now test if the method navigates correctly to the DetailsPage.
+
+``` C#
+    [Fact]
+    public async void GoToCatDetailsAsync()
+    {
+        var connectivtyMock = GetConnectivtyMock();
+        var geolocationMock = GetGeolocationMock();
+        var catServiceMock = GetCatServiceMock();
+        var catsViewModel = new CatsViewModel(catServiceMock.Object, connectivtyMock.Object, geolocationMock.Object);
+        var cat = new Cat(string.Empty, string.Empty, string.Empty, string.Empty, 0, 0, 0);
+        var appMock = new Mock<App>();
+        Application.Current = appMock.Object;
+
+        await catsViewModel.GoToCatDetailsAsync(cat);
+
+        Shell.Current.CurrentPage.Should().BeOfType(typeof(DetailsPage));
+    }
+ ```
+
+The test above has one crucial aspect to pay attention to if you want to use the Shell inside the tests, namely the app mock.
+The test setup does not configure the complete MAUI app. Therefore, crucial features like the Shell are only usable when you set them up yourself.
```

Comment:
Suggestion: crucial features -> platform dependent features?

Reply:
Good point, changed it.